lms/djangoapps/course_api/views.py

```diff
@@ -201,9 +210,24 @@ def get_queryset(self):
         if not form.is_valid():
             raise ValidationError(form.errors)
 
-        return list_courses(
+        db_courses = list_courses(
             self.request,
             form.cleaned_data['username'],
             org=form.cleaned_data['org'],
             filter_=form.cleaned_data['filter_'],
         )
+
+        if not settings.FEATURES['ENABLE_COURSEWARE_SEARCH'] or not form.cleaned_data['search_term']:
```

Comment:
👍 Thank you for adding the Feature Flag.

Reply:
@staubina Just to be clear. This more of reusing an existing feature flag. It's easy to make up a new one though.

Reply:
I will consult with Product when I talk with them tomorrow.

Reply:
A short update, I am looking now to just verify where this setting is used elsewhere to ensure we will not enable this on production by accident.

Reply:
Thanks @staubina, let me know if it's better to add a new feature for it.

Reply:
I did some digging and it seems acceptable to use this Flag in combination with the search_term for this api.